src/control/cmd/dmg/storage.go

```diff
@@ -286,3 +290,58 @@ func groupFormatResults(results client.StorageFormatResults, summary bool) (grou
 
 	return
 }
+
+// prepareCmdDisplay returns tabulated output of grouped host prepare results.
+func prepareCmdDisplay(results *client.StoragePrepareResp) (string, error) {
+	out := &bytes.Buffer{}
+
+	groups, err := groupPrepareResults(results)
+	if err != nil {
+		return "", err
+	}
+
+	if len(groups) > 0 {
+		fmt.Fprintf(out, "\n%s\n", groups)
+	}
+
+	return formatHostGroups(out, groups), nil
+}
+
+// groupPrepareResults collects identical output keyed on hostset from
+// prepare results and returns grouped output.
+func groupPrepareResults(resp *client.StoragePrepareResp) (groups hostlist.HostGroups, err error) {
+	var host string
+	buf := &bytes.Buffer{}
+	groups = make(hostlist.HostGroups)
+
+	for _, srv := range resp.Servers {
+		buf.Reset()
+
+		host, _, err = common.SplitPort(srv, 0) // disregard port when grouping output
+		if err != nil {
+			return
+		}
+
+		sres := resp.Scm[srv]
+		switch {
+		case sres.Err != nil:
+			fmt.Fprintf(buf, "SCM Error: %s\n", sres.Err)
+		case len(sres.Namespaces) > 0:
+			fmt.Fprintf(buf, "%s\n", scmNsScanTable(sres.Namespaces))
+		default:
+			fmt.Fprintf(buf, "SCM ???\n")
```

Comment:
I think it would be better to say something like "SCM Unknown State" or something like that